File: backend/controllers/lesson_controller.py
from flask import request, jsonify, g
from controllers import handle_errors
import services.lesson_service as svc


@handle_errors
def get_lessons():
    q = request.args.get("q", "").strip()
    return jsonify({"success": True, "data": svc.get_lessons(g.user["user_id"], q)})


@handle_errors
def get_lesson(lesson_id):
    return jsonify({"success": True, "data": svc.get_lesson(lesson_id)})


@handle_errors
def create_lesson():
    title = (request.json or {}).get("title", "").strip()
    if not title:
        return jsonify({"success": False, "message": "Tiêu đề không được trống"}), 400
    svc.create_lesson(g.user["user_id"], title)
    return jsonify({"success": True, "message": "Tạo bài học thành công"})


@handle_errors
def update_lesson(lesson_id):
    title = (request.json or {}).get("title", "").strip()
    if not title:
        return jsonify({"success": False, "message": "Tiêu đề không được trống"}), 400
    svc.update_lesson(lesson_id, g.user["user_id"], title)
    return jsonify({"success": True, "message": "Cập nhật bài học thành công"})


@handle_errors
def delete_lesson(lesson_id):
    svc.delete_lesson(lesson_id, g.user["user_id"])
    return jsonify({"success": True, "message": "Đã xóa bài học"})

File: backend/src/controllers/lesson.controller.js
import * as svc from '../services/lesson.service.js';
import { success, created, successMsg } from '../utils/response.js';

export const getLessons = async (req, res) => {
  const q = req.query.q || '';
  const page = parseInt(req.query.page) || 1;
  const limit = parseInt(req.query.limit) || 10;
  success(res, await svc.getLessons(req.user.id, q, page, limit));
};

export const getLesson = async (req, res) => {
  success(res, await svc.getLesson(parseInt(req.params.id)));
};

export const createLesson = async (req, res) => {
  const { title } = req.body;
  await svc.createLesson(req.user.id, title.trim());
  created(res, null, 'Tạo bài học thành công');
};

export const updateLesson = async (req, res) => {
  const { title } = req.body;
  await svc.updateLesson(parseInt(req.params.id), req.user.id, title.trim());
  successMsg(res, 'Cập nhật thành công');
};

export const deleteLesson = async (req, res) => {
  await svc.deleteLesson(parseInt(req.params.id), req.user.id);
  successMsg(res, 'Đã xóa bài học');
};
